lib/rules/complexity.js

```diff
@@ -45,6 +45,10 @@ module.exports = {
                             max: {
                                 type: "integer",
                                 minimum: 0
+                            },
+                            variant: {
+                                type: "string",
+                                enum: ["classic", "modified"]
```

Comment:
By the way why can't we use the `default` property in JSONSchema? 
If I add e.g. `default: "classic"` here, all the tests fail with:

```
Error: Schema for rule complexity is invalid: default is ignored for: data1.variant
```

Would be nice to specify defaults at schema level

Reply:
Because `default` doesn't work the way people think it does in ESLint. We're working on an alternate way to deal with default options: https://github.com/eslint/eslint/pull/17656

Reply:
As in, the current option schema validator merely _validates_ the input against the schema and does not fill in the default values?